### PR TITLE
refactor: move heartbeat configuration into an independent section

### LINKED ISSUE #1936: Refactor: move heartbeat configuration into an independent section in config file
### What type of enhancement is this?

Configuration, User experience

### What does the enhancement do?

The PR #1864 has introduced two new parameters to control heartbeat:

```toml
# Interval for sending heartbeat task to the Metasrv in milliseconds, 5000 by default.
heartbeat_interval_millis = 5000
# Interval for retry sending heartbeat task in milliseconds, 5000 by default.
retry_interval_millis = 5000
```

I think they must be moved into an independent section of the config file such as `meta` or `heartbeat`:

```toml
[meta]
# Interval for sending heartbeat task to the Metasrv in milliseconds, 5000 by default.
heartbeat_interval_millis = 5000
# Interval for retry sending heartbeat task in milliseconds, 5000 by default.
retry_interval_millis = 5000
```

And they don't need to be present in standalone mode.

### Implementation challenges

_No response_

### PR DESCRIPTION
**Pull request by Kree0**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Add a new HeartBeatOptions struct, use this instead heartbeat_interval_millis and retry_interval_millis, change HeartbeatTask::new() to take it as argument.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
close #1936

### CODE REVIEW

**Comment by killme2008** [comment]
Thanks for your contribution! Great work.

We have to modify the example configuration in 

https://github.com/GreptimeTeam/greptimedb/blob/97cfa3d6c97f4a4ad3413a20e29b5f0859e304fc/config/datanode.example.toml#L14

and

https://github.com/GreptimeTeam/greptimedb/blob/97cfa3d6c97f4a4ad3413a20e29b5f0859e304fc/config/frontend.example.toml#L4

**Comment by etolbakov** [comment]
@Kree0 I believe [integration tests](https://github.com/GreptimeTeam/greptimedb/blob/97cfa3d6c97f4a4ad3413a20e29b5f0859e304fc/tests-integration/tests/http.rs#L514) should be modified as well.

**Comment by MichaelScofield** [comment]
@Kree0 Yes, `heartbeat_interval_millis` in datanode config file should be placed under `[heartbeat]` section as well.

**Comment by etolbakov** [comment]
Just would like to share the observation.
Since the new section is introduced maybe it makes sense to shorten the name of the setting, i.e.

```toml
[heartbeat]
# Interval for sending heartbeat messages to the Metasrv in milliseconds, 5000 by default.
interval_millis = 5000
```
what do you think about it?

File: src/frontend/src/heartbeat.rs [hunk 1]
```diff
@@ -32,6 +33,22 @@ use crate::error::Result;
 
 pub mod handler;
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(default)]
+pub struct HeartBeatOptions {
```
Comment (etolbakov) [inline src/frontend/src/heartbeat.rs:38]: Also, I've noticed that you define 2 similar structs for `heartbeat`. I wonder if should we go with a single one?
nit: I would suggest the following renaming `HeartBeatOptions` -> `HeartbeatOptions`

Comment (killme2008) [reply src/frontend/src/heartbeat.rs:38]: We can move `HeartBeatOptions` to `server` crate and keep two fields existing.

And let `frontend` and `datanode` use this option.

What do u think? @MichaelScofield

File: src/servers/src/heartbeat_options.rs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+use serde::{Deserialize, Serialize};
```
Comment (etolbakov) [inline src/servers/src/heartbeat_options.rs:1]: @Kree0 could you please add the licence header here? you can copy it from any existing file

File: tests-integration/tests/http.rs [hunk 3]
```diff
@@ -511,7 +511,9 @@ pub async fn test_config_api(store_type: StorageType) {
     enable_memory_catalog = false
     rpc_addr = "127.0.0.1:3001"
     rpc_runtime_size = 8
-    heartbeat_interval_millis = 5000
+
+    [heartbeat]
+    interval_millis = 5000
```
Comment (etolbakov) [inline tests-integration/tests/http.rs:516]: @Kree0 looks like one more adjustment is required for this test.
To reproduce the failure you can use `cargo test test_config_api` command

**Comment by etolbakov** [comment]
@Kree0 your change LGTM.  Let's see what other guys think.
not sure if the `test_schedule_duplicate_tasks` is a flaky one.

Also a side note about commit messages: it's always good to have a descriptive message that summarises the change even if it's trivial. It helps to understand the necessity of a particular commit.

**Pull request closed without merging**